src/store/reducer/priceBarSlice.ts

```diff
@@ -0,0 +1,51 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { RootState } from "..";
+
+export interface PriceBar {
+  color: string;
+  quantity: number;
+}
+
+export interface PriceBarState {
+  blocks: PriceBar[];
+}
+export const initialState: PriceBarState = {
+  blocks: [
+    {
+      color: "#72818B",
+      quantity: 1277,
+    },
+    {
+      color: "#B61313",
+      quantity: 2576,
+    },
+    {
+      color: "#195955",
+      quantity: 1098,
+    },
+    {
+      color: "#2C4E8A",
+      quantity: 637,
+    },
+    {
+      color: "#606F14",
+      quantity: 130,
+    },
+  ],
+};
+
+export const priceBarSlice = createSlice({
+  name: "priceBar",
+  initialState,
+  reducers: {
+    changeTileQuantity: (state, action: PayloadAction<PriceBar[]>) => {
+      state.blocks = action.payload;
+    },
+  },
+});
+
+export const { changeTileQuantity } = priceBarSlice.actions;
+
+export const priceBarData = (state: RootState) => state.priceBar;
+
+export default priceBarSlice.reducer;
```

Comment:
I got confused. Is there any relationship between this file and undo and redo function?

Reply:
No, there is no relationship between priceBarSlice and undo/redo function. I separated the priceBarSlice from tileSlice, and only add undo/redo function to tileSlice, as I only need the tileSlice to be undoable. If I do not separate them, when undoing the color of court, the redux-undo "future" will be overwritten by the update of pricebar state.

Reply:
Make sense. Nice work.